File: pset6/caesar.py
from cs50 import get_string
import sys

# Finding correct key
digit = 0
argc = len(sys.argv)

if (argc == 2):

    # Get key argument
    key = sys.argv[1]

    # Check if key is a number
    if (key.isdigit() == False):
        print("Usage: ./caesar key")
        sys.exit(1)
    else:
        # Convert key into an integer
        key = int(sys.argv[1])

        # Asking Plain text
        p_tx = get_string("plaintext: ")

        # Declaring ciphertext string
        cx_tx = []

        # length of plain text
        txt_len = len(p_tx)

        # Shifting letters in the string
        for j in range(txt_len):

            # Shifting lowercase alphabets
            if ((p_tx[j] >= 'a') and (p_tx[j] <= 'z')):
                count = ord(p_tx[j])
                chk_count = (count - 1)
                ans = count + key

                while(count <= ans):
                    if (chk_count == 122):
                        chk_count = 97
                        count += 1
                    else:
                        chk_count += 1
                        count += 1

                cx_tx.append(chr(chk_count))

            # Shifting uppercase alphabets
            elif ((p_tx[j] >= 'A') and (p_tx[j] <= 'Z')):
                count = ord(p_tx[j])
                chk_count = (count - 1)
                ans = count + key

                while(count <= ans):
                    if (chk_count == 90):
                        chk_count = 65
                        count += 1
                    else:
                        chk_count += 1
                        count += 1

                cx_tx.append(chr(chk_count))

            else:
                cx_tx.append(p_tx[j])

        print("ciphertext: ", end="")
        for i in cx_tx:
            print(i, end="")
        print()

else:
    print("Usage: ./caesar key")
    sys.exit(1)

File: pset2/caesar.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>
#include <math.h>

int main(int argc, string argv[])
{
    // Finding correct key
    int digit = 0;
    if (argc == 2)
    {
        // Getting input string
        string s = argv[1];

        // Finding the length of the string
        int c = strlen(s);

        // split input string into digits
        char chr[c];
        int dig_unit = 0;

        // Checking for numeric digits
        for (int i = 0; i < c; i++)
        {
            chr[i] = s[i];
            if (isdigit(chr[i]) == 0)
            {
                printf("Usage: ./caesar key\n");
                return 1;
            }
            else
            {
                dig_unit = (int)(chr[i] - '0');
                dig_unit *= pow(10, c - i - 1);
                digit = dig_unit + digit;
            }
            printf("%c\n",chr[i]);
        }
    }
    else
    {
        printf("Usage: ./caesar key\n");
        return 1;
    }
    printf("%i\n",digit);
    // Asking Plain text
    string p_tx = get_string("plaintext: ");
    // Shifting letters in the string
    for (int j = 0; p_tx[j] != '\0'; j++)
    {
        // Shifting lowercase alphabets
        if ((p_tx[j] >= 'a') && (p_tx[j] <= 'z'))
        {
            int count = (int) p_tx[j];
            int chk_count = count - 1;
            int ans = count + digit;
            for (int k = count; count <= ans; count++)
            {
                if (chk_count == 122)
                {
                    chk_count = 97;
                }
                else
                {
                    chk_count++;
                }
            }
            p_tx[j] = (char) chk_count;
        }
        // Shifting uppercase alphabets
        else if ((p_tx[j] >= 'A') && (p_tx[j] <= 'Z'))
        {
            int count = (int) p_tx[j];
            int chk_count = count - 1;
            int ans = count + digit;
            for (int k = count; count <= ans; count++)
            {
                if (chk_count == 90)
                {
                    chk_count = 65;
                }
                else
                {
                    chk_count++;
                }
            }
            p_tx[j] = (char) chk_count;
        }
    }
    printf("ciphertext: %s\n", p_tx);
}

